tests/gold_tests/session_sharing/session_management_client.py

```diff
@@ -0,0 +1,60 @@
+'''
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import requests
+import argparse
+import logging
+
+
+def main():
+    """
+    Make three HTTP transactions across two connections.
+
+    This client helps verify session reuse functionality.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--port", "-p",
+                        type=int,
+                        help="Port to use")
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.DEBUG, format="%(message)s")
+
+    # Make a request on connection 1.
+    firstConnection = requests.Session()
+    firstConnection.get(
+        'http://127.0.0.1:{0}/one'.format(args.port),
+        headers={'Connection': "keep-alive",
```

Comment:
1. You shouldn't need `Connection: keep-alive` for HTTP/1.1.  Could test using both.
2. It looks like you are having the client close the connection. You can have keep-alive/close independently on the client without effecting the origin, and the reverse is true.

Reply:
Thanks Bryan. This is helpful.

I removed the explicit keep-alives since they are not necessary, as you point out. microserver does the right thing and assumes a persistent connection.

I also simplified the test to simply use curl as the client since it can close the connection on the client side with each transaction and the server-side connection persists (and gets reused) as desired.